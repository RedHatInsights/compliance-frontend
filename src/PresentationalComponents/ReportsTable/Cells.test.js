import {
  Name,
  OperatingSystem,
  CompliantSystems,
  PDFExportDownload,
} from './Cells';

describe('Name', () => {
  it('expect to render without error', () => {
    const wrapper = shallow(
      <Name
        {...{
          id: 'ID',
          name: 'NAME',
          policyType: 'POLICY_TYPE',
          policy: {
            id: 'POLICY_ID',
            name: 'POLICY_NAME',
          },
        }}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('OperatingSystem', () => {
  const defaultProps = {
    osMajorVersion: '7',
  };

  it('expect to render without error', () => {
    const wrapper = shallow(<OperatingSystem {...defaultProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('expect to render with SSG version', () => {
    const wrapper = shallow(
      <OperatingSystem
        {...defaultProps}
        benchmark={{ version: '1.2.3' }}
        policy={null}
        unsupportedHostCount={0}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('expect to render with unsupported warning', () => {
    const wrapper = shallow(
      <OperatingSystem
        {...defaultProps}
        benchmark={{ version: '1.2.3' }}
        unsupportedHostCount={3}
        policy={null}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('CompliantSystems', () => {
  const deftaultProps = {
    testResultHostCount: 10,
    compliantHostCount: 9,
  };

  it('expect to render without error', () => {
    const wrapper = shallow(<CompliantSystems {...deftaultProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('expect to render with unsupported hosts', () => {
    const wrapper = shallow(
      <CompliantSystems {...deftaultProps} unsupportedHostCount={42} />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('PDFExportDownload', () => {
  it('expect to render without error', () => {
    const wrapper = shallow(<PDFExportDownload id="ID1" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
