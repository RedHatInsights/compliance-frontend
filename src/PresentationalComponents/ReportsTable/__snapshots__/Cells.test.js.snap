// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompliantSystems expect to render with unsupported hosts 1`] = `
<Fragment>
  <ReportChart
    chartClass="report-table-chart-container"
    hasLegend={false}
    profile={
      Object {
        "compliantHostCount": 9,
        "testResultHostCount": 10,
        "unsupportedHostCount": 42,
      }
    }
  />
</Fragment>
`;

exports[`CompliantSystems expect to render without error 1`] = `
<Fragment>
  <ReportChart
    chartClass="report-table-chart-container"
    hasLegend={false}
    profile={
      Object {
        "compliantHostCount": 9,
        "testResultHostCount": 10,
      }
    }
  />
</Fragment>
`;

exports[`Name expect to render without error 1`] = `
<TextContent>
  <LinkWithPermission
    style={
      Object {
        "marginRight": ".5rem",
      }
    }
    to="/reports/ID"
  >
    POLICY_NAME
  </LinkWithPermission>
  <PolicyPopover
    position="right"
    profile={
      Object {
        "id": "ID",
        "name": "NAME",
        "policy": Object {
          "id": "POLICY_ID",
          "name": "POLICY_NAME",
        },
        "policyType": "POLICY_TYPE",
      }
    }
  />
  <GreySmallText>
    POLICY_TYPE
  </GreySmallText>
</TextContent>
`;

exports[`OperatingSystem expect to render with SSG version 1`] = `
<Fragment>
  RHEL 
  7
  <Text>
    <GreySmallText>
      SSG: 1.2.3
    </GreySmallText>
  </Text>
</Fragment>
`;

exports[`OperatingSystem expect to render with unsupported warning 1`] = `
<Fragment>
  RHEL 
  7
  <Text>
    <GreySmallText>
      <UnsupportedSSGVersion>
        SSG: 1.2.3
      </UnsupportedSSGVersion>
    </GreySmallText>
  </Text>
</Fragment>
`;

exports[`OperatingSystem expect to render without error 1`] = `
<Fragment>
  RHEL 
  7
</Fragment>
`;

exports[`PDFExportDownload expect to render without error 1`] = `
<BackgroundLink
  className="pf-u-mr-md"
  component={[Function]}
  ouiaId="ReportsDownloadReportPDFLink"
  to="/reports/ID1/pdf"
  variant="plain"
>
  <DownloadIcon
    color="currentColor"
    noVerticalAlign={false}
    size="sm"
  />
</BackgroundLink>
`;
